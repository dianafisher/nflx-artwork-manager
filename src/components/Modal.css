.backdrop {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(0,0,0,0.3);
  padding: 50px;
}

.modal {
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.2), 0 1px 5px 0 rgba(0,0,0,.12);
}

.fullsize-image {
  width: 960px;
  height: 512px;
}

.title {
  background-color: #df2230; /*#00bcd4;*/
  height: 48px;
  font-size: 20px;
  color: #fff;
  display: flex;
  align-items: center;
  padding-left: 20px;
}

.detail {
  text-align: left;
  padding-left: 20px;
}

.detail span {
  padding-left: 20px;
}

.close-button {
  background-color: #df2230;
  color: rgba(255,255,255,.84);
  border-radius: 5px;
  box-shadow: 0 2px 2px 0 rgba(0,0,0,.14), 0 3px 1px -2px rgba(0,0,0,.2), 0 1px 5px 0 rgba(0,0,0,.12);
  padding: 8px 30px;
  margin: 10px 1px;
  font-size: 14px;
  text-transform: uppercase;
  border: none;
  transition: box-shadow .2s cubic-bezier(.4,0,1,1),background-color .2s cubic-bezier(.4,0,.2,1),color .2s cubic-bezier(.4,0,.2,1);
}

.close-button:focus {
  outline: 0;
}

.close-button:hover{
  background-color: #f55549;
}
