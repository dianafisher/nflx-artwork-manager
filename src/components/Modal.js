import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Modal.css';

class Modal extends Component {

  render() {
    // Render nothing if the "show" prop is false
    if(!this.props.show) {
      return null;
    }

    return (
      <div className="backdrop">
        <div className="modal">
          <div className='title'>{this.props.item.movieName}</div>
          <img
            className="fullsize-image"
            src={this.props.item.fullSizeImageUrl}
            alt='full size'
          ></img>
          <div className="detail">
            <label>Movie ID:</label>
            <span>{this.props.item.movieId}</span>
          </div>
          <div className="detail">
            <label>Image Type:</label>
            <span>{this.props.item.imageType}</span>
          </div>
          <div className="detail">
            <label>Language:</label>
            <span>{this.props.item.language}</span>
          </div>
          <div className="detail">
            <label>Language Code:</label>
            <span>{this.props.item.languageCode}</span>
          </div>
          <div className="footer">
            <button
              className="close-button"
              onClick={this.props.onClose}>
              Close
            </button>
          </div>
        </div>
      </div>
    );
  }
}

Modal.propTypes = {
  onClose: PropTypes.func.isRequired,
  show: PropTypes.bool,
  item: PropTypes.object.isRequired
};

export default Modal;
