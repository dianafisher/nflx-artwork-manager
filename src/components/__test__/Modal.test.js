import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../Modal';

const item = {
  movieName: 'Test Movie',
  fullSizeImageUrl: 'http://via.placeholder.com/350x150',
  movieId: 1,
  imageType: 'sdp',
  language: 'English',
  languageCode: 'en'
};

describe('Modal', () => {
  
  it('renders nothing when closed', () => {
    expect(shallow(
      <Modal item={item} onClose={jest.fn()}/>
    )).toMatchSnapshot();
  });

  it('renders when open', () => {
    expect(shallow(
      <Modal item={item} show={true} onClose={jest.fn()}/>
    )).toMatchSnapshot();
  });

  it('calls onClose when button is clicked', () => {
    const onClose = jest.fn();
    const wrapper = shallow(
      <Modal item={item} show={true} onClose={onClose}/>
    );

    wrapper.find('button').simulate('click');
    expect(onClose).toBeCalled();
  });
});
