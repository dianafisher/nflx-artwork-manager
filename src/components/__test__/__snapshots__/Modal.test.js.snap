// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders nothing when closed 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": null,
  "nodes": Array [
    null,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Modal
        item={
                Object {
                        "fullSizeImageUrl": "http://via.placeholder.com/350x150",
                        "imageType": "sdp",
                        "language": "English",
                        "languageCode": "en",
                        "movieId": 1,
                        "movieName": "Test Movie",
                      }
        }
        onClose={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Modal {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "fullSizeImageUrl": "http://via.placeholder.com/350x150",
            "imageType": "sdp",
            "language": "English",
            "languageCode": "en",
            "movieId": 1,
            "movieName": "Test Movie",
          },
          "onClose": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": null,
        "_debugID": 2,
        "_renderedOutput": null,
      },
      "_renderedNodeType": 2,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Modal
    item={
        Object {
            "fullSizeImageUrl": "http://via.placeholder.com/350x150",
            "imageType": "sdp",
            "language": "English",
            "languageCode": "en",
            "movieId": 1,
            "movieName": "Test Movie",
          }
    }
    onClose={[Function]}
/>,
}
`;

exports[`Modal renders when open 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="backdrop"
>
    <div
        className="modal"
    >
        <div
            className="title"
        >
            Test Movie
        </div>
        <img
            alt="full size"
            className="fullsize-image"
            src="http://via.placeholder.com/350x150"
        />
        <div
            className="detail"
        >
            <label>
                Movie ID:
            </label>
            <span>
                1
            </span>
        </div>
        <div
            className="detail"
        >
            <label>
                Image Type:
            </label>
            <span>
                sdp
            </span>
        </div>
        <div
            className="detail"
        >
            <label>
                Language:
            </label>
            <span>
                English
            </span>
        </div>
        <div
            className="detail"
        >
            <label>
                Language Code:
            </label>
            <span>
                en
            </span>
        </div>
        <div
            className="footer"
        >
            <button
                className="close-button"
                onClick={[Function]}
            >
                Close
            </button>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="backdrop"
>
      <div
            className="modal"
      >
            <div
                  className="title"
            >
                  Test Movie
            </div>
            <img
                  alt="full size"
                  className="fullsize-image"
                  src="http://via.placeholder.com/350x150"
            />
            <div
                  className="detail"
            >
                  <label>
                        Movie ID:
                  </label>
                  <span>
                        1
                  </span>
            </div>
            <div
                  className="detail"
            >
                  <label>
                        Image Type:
                  </label>
                  <span>
                        sdp
                  </span>
            </div>
            <div
                  className="detail"
            >
                  <label>
                        Language:
                  </label>
                  <span>
                        English
                  </span>
            </div>
            <div
                  className="detail"
            >
                  <label>
                        Language Code:
                  </label>
                  <span>
                        en
                  </span>
            </div>
            <div
                  className="footer"
            >
                  <button
                        className="close-button"
                        onClick={[Function]}
                  >
                        Close
                  </button>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Modal
        item={
                Object {
                        "fullSizeImageUrl": "http://via.placeholder.com/350x150",
                        "imageType": "sdp",
                        "language": "English",
                        "languageCode": "en",
                        "movieId": 1,
                        "movieName": "Test Movie",
                      }
        }
        onClose={[Function]}
        show={true}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Modal {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "item": Object {
            "fullSizeImageUrl": "http://via.placeholder.com/350x150",
            "imageType": "sdp",
            "language": "English",
            "languageCode": "en",
            "movieId": 1,
            "movieName": "Test Movie",
          },
          "onClose": [Function],
          "show": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="backdrop"
>
          <div
                    className="modal"
          >
                    <div
                              className="title"
                    >
                              Test Movie
                    </div>
                    <img
                              alt="full size"
                              className="fullsize-image"
                              src="http://via.placeholder.com/350x150"
                    />
                    <div
                              className="detail"
                    >
                              <label>
                                        Movie ID:
                              </label>
                              <span>
                                        1
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Image Type:
                              </label>
                              <span>
                                        sdp
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Language:
                              </label>
                              <span>
                                        English
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Language Code:
                              </label>
                              <span>
                                        en
                              </span>
                    </div>
                    <div
                              className="footer"
                    >
                              <button
                                        className="close-button"
                                        onClick={[Function]}
                              >
                                        Close
                              </button>
                    </div>
          </div>
</div>,
        "_debugID": 4,
        "_renderedOutput": <div
          className="backdrop"
>
          <div
                    className="modal"
          >
                    <div
                              className="title"
                    >
                              Test Movie
                    </div>
                    <img
                              alt="full size"
                              className="fullsize-image"
                              src="http://via.placeholder.com/350x150"
                    />
                    <div
                              className="detail"
                    >
                              <label>
                                        Movie ID:
                              </label>
                              <span>
                                        1
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Image Type:
                              </label>
                              <span>
                                        sdp
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Language:
                              </label>
                              <span>
                                        English
                              </span>
                    </div>
                    <div
                              className="detail"
                    >
                              <label>
                                        Language Code:
                              </label>
                              <span>
                                        en
                              </span>
                    </div>
                    <div
                              className="footer"
                    >
                              <button
                                        className="close-button"
                                        onClick={[Function]}
                              >
                                        Close
                              </button>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Modal
    item={
        Object {
            "fullSizeImageUrl": "http://via.placeholder.com/350x150",
            "imageType": "sdp",
            "language": "English",
            "languageCode": "en",
            "movieId": 1,
            "movieName": "Test Movie",
          }
    }
    onClose={[Function]}
    show={true}
/>,
}
`;
